### PR TITLE
Add var to force docker save

### PR DESCRIPTION
**Pull request by keithmattix**
Couldn't see another way to make docker export to tarballs without needlessly setting a dummy DOCKER_PASSWORD

### CODE REVIEW

File: ci/run_envoy_docker.sh [hunk 1]
```diff
@@ -126,7 +126,7 @@ docker run --rm \
        -e CI_TARGET_BRANCH \
        -e DOCKERHUB_USERNAME \
        -e DOCKERHUB_PASSWORD \
-       -e DOCKER_SAVE_IMAGE \
+       -e ENVOY_DOCKER_SAVE_IMAGE \
```
Comment (phlax) [inline ci/run_envoy_docker.sh:129]:
```suggestion
       -e ENVOY_DOCKER_LOAD_IMAGES \
```

**Review by phlax — DISMISSED** [review]
lgtm, altho i confess im no less confused if this fixes 8/

i would like to get final confirmation it does before landing

/wait

**Review by phlax — APPROVED** [review]
lgtm, thanks @keithmattix 

apologies for the confusion

**Comment by keithmattix** [comment]
Need to cherry-pick to 1.26 and 1.27

**Pull request merged**